deltaconcor/src/main/java/uk/gov/justice/laadces/deltaconcor/Runner.java

```diff
@@ -0,0 +1,353 @@
+            }
+        }
+        log.info("run: {} maatId predecessors found", maatIdToContribution.size());
+
+        generateDailyCounts(START_DATE, startId, endId);
+        log.info("run: Counts generated, now outputting CSV files");
+
+        csvoutput.writeChanges("/tmp/deltaconcor-new.csv", newChanges);
+        csvoutput.writeChanges("/tmp/deltaconcor-upd.csv", updChanges);
+        csvoutput.writeChangeLogs("/tmp/deltaconcor-log.csv", changeLogs);
+        log.info("run: Done");
+    }
+
+    private void reset(final LocalDate startDate, final LocalDate endDate) {
+        final int nDays = (int) ChronoUnit.DAYS.between(startDate, endDate);
+        log.info("reset: {} days in period [{}, {})", nDays, startDate, endDate);
+        maatIdsToBackFill.clear();
+        maatIdToContribution.clear();
+        newChanges.clear();
+        updChanges.clear();
+        newChanges.trimToSize();
+        updChanges.trimToSize();
+        newChanges.ensureCapacity(nDays);
+        updChanges.ensureCapacity(nDays);
+        for (int i = 0; i < nDays; ++i) {
+            newChanges.add(new Change(startDate.plusDays(i).toString()));
+            updChanges.add(new Change(startDate.plusDays(i).toString()));
+        }
+        changeLogs.clear();
+        changeLogs.trimToSize();
+        changeLogs.ensureCapacity(nDays); // will actually be a large multiple of nDays
+    }
+
+    private long findMaatIdsToBackFill(final long startId, final long endId) {
+        return visitForwards(startId, endId, concorContribution -> {
+            ++countRowsInDateRange;
+            final long maatId = concorContribution.repId();
+            final CONTRIBUTIONS dto = xmlTransformService.fromXML(concorContribution.fullXml());
+            if (dto == null) {
+                log.warn("findMaatIdsToBackFill: Failed to transform XML for concorId {}, maatId {}", concorContribution.id(), maatId);
+            } else if (dto.getFlag() == null) {
+                log.warn("findMaatIdsToBackFill: No flag for concorId {}, maatId {}", concorContribution.id(), maatId);
+            } else if (dto.getFlag().equalsIgnoreCase("new")) {
+                if (maatIdToContribution.containsKey(maatId)) {
+                    log.warn("findMaatIdsToBackFill: More than one 'new' for concorId {}, maatId {}", concorContribution.id(), maatId);
+                } else {
+                    maatIdToContribution.put(maatId, dto);
+                }
+            } else if (dto.getFlag().equalsIgnoreCase("update")) {
+                if (!maatIdToContribution.containsKey(maatId)) {
+                    if (!maatIdsToBackFill.contains(maatId)) {
+                        maatIdsToBackFill.add(maatId);
+                    } else {
+                        ++countRows1;
+                    }
+                } else {
+                    ++countRows2;
```

Comment:
Same as above, can it be given a more descriptive name? Is it countUpdatesToNewData?

Reply:
Ditto